libtransact/src/contract/address/triple_key_hash.rs

```diff
@@ -25,14 +25,14 @@ impl TripleKeyHashAddresser {
         prefix: String,
         first_hash_length: Option<usize>,
         second_hash_length: Option<usize>,
-    ) -> TripleKeyHashAddresser {
+    ) -> Result<TripleKeyHashAddresser, AddresserError> {
         let (first, second) =
-            calculate_hash_lengths(prefix.len(), first_hash_length, second_hash_length);
-        TripleKeyHashAddresser {
+            calculate_hash_lengths(prefix.len(), first_hash_length, second_hash_length)?;
+        Ok(TripleKeyHashAddresser {
```

Comment:
A lot could potentially be simplified here by making first_hash_length,second_hash_length required fields or filling them on during construction (as opposed to keeping them Options in the struct).

Reply:
They aren't options in the struct, this method is just parsing through the optional arguments from the constructor. Ultimately, the Addresser is doesn't have the hash lengths as Options.